cmd/agent/subcommands/secret/command.go

```diff
@@ -12,56 +12,67 @@ import (
 	"os"
 
 	"github.com/spf13/cobra"
+	"go.uber.org/fx"
 
 	"github.com/DataDog/datadog-agent/cmd/agent/command"
-	"github.com/DataDog/datadog-agent/cmd/agent/common"
+	"github.com/DataDog/datadog-agent/comp/core"
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	"github.com/DataDog/datadog-agent/comp/core/log"
 	"github.com/DataDog/datadog-agent/pkg/api/util"
-	"github.com/DataDog/datadog-agent/pkg/config"
+	pkgconfig "github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/secrets"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
 )
 
+// cliParams are the command-line arguments for this subcommand
+type cliParams struct {
+	*command.GlobalParams
+}
+
 // Commands returns a slice of subcommands for the 'agent' command.
 func Commands(globalParams *command.GlobalParams) []*cobra.Command {
+	cliParams := &cliParams{
+		GlobalParams: globalParams,
+	}
 	secretInfoCommand := &cobra.Command{
 		Use:   "secret",
 		Short: "Print information about decrypted secrets in configuration.",
 		Long:  ``,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			err := common.SetupConfigWithoutSecrets(globalParams.ConfFilePath, "")
-			if err != nil {
-				fmt.Printf("unable to set up global agent configuration: %v\n", err)
-				return nil
-			}
-
-			err = config.SetupLogger(config.CoreLoggerName, config.GetEnvDefault("DD_LOG_LEVEL", "off"), "", "", false, true, false)
-			if err != nil {
-				fmt.Printf("Cannot setup logger, exiting: %v\n", err)
-				return err
-			}
-
-			if err := util.SetAuthToken(); err != nil {
-				fmt.Println(err)
-				return nil
-			}
-
-			if err := showSecretInfo(); err != nil {
-				fmt.Println(err)
-				return nil
-			}
-			return nil
+			return fxutil.OneShot(secret,
+				fx.Supply(cliParams),
+				fx.Supply(core.BundleParams{
+					ConfFilePath:      globalParams.ConfFilePath,
+					ConfigLoadSecrets: false,
+				}.LogForOneShot("CORE", "off", true)),
+				core.Bundle,
+			)
 		},
 	}
 
 	return []*cobra.Command{secretInfoCommand}
 }
 
-func showSecretInfo() error {
+func secret(log log.Component, config config.Component, cliParams *cliParams) error {
+	if err := util.SetAuthToken(); err != nil {
+		fmt.Println(err)
+		return nil
+	}
+
+	if err := showSecretInfo(config); err != nil {
+		fmt.Println(err)
+		return nil
+	}
+	return nil
+}
+
+func showSecretInfo(config config.Component) error {
 	c := util.GetClient(false)
-	ipcAddress, err := config.GetIPCAddress()
+	ipcAddress, err := pkgconfig.GetIPCAddress()
```

Comment:
Do we plan to move this to the config interface in a later PR ?

Reply:
I'm not sure -- there's a _lot_ of purpose-specific stuff in `pkg/config` and I'd rather not add all of that to comp/core/config.  I can think of a few ways to address this but haven't made any decisions yet.